contracts/truefi/LoanToken.sol

```diff
@@ -373,13 +373,19 @@ contract LoanToken is ILoanToken, ERC20 {
 
     /**
      * @dev Internal function for loan repayment
+     * If _amount is sufficient, then this also settles the loan
      * @param _sender account sending currencyToken to repay
      * @param _amount amount of currencyToken to repay
      */
     function _repay(address _sender, uint256 _amount) internal onlyAfterWithdraw {
-        require(_amount <= debt.sub(_balance()), "LoanToken: Cannot repay over the debt");
+        uint256 outstandingDebt = debt.sub(_balance());
+        require(_amount <= outstandingDebt, "LoanToken: Cannot repay over the debt");
         emit Repaid(_sender, _amount);
+
         currencyToken.safeTransferFrom(_sender, address(this), _amount);
+        if (_amount >= outstandingDebt) {
```

Comment:
instead of checking amount, check isRepaid(). This will settle the loan if the total debt is repaid, not just the amount paid in this transaction

Reply:
Thanks for the catch! You're right, I should have gone for clarity over saving an extra isRepaid() function call.

I'll also note that calling settle() after the currency transfer violates the checks-effects-interactions pattern, but I think it's justified here because:
1) the only state change after the transfer is status = Settled, which doesn't care about reentrancy. (Currently we call settle() a long time after repaying anyway.)
2) I think it's dangerous to remove the balance >= debt check from settle(), which we'd need to do if we shifted the call to settle() up before the transfer.
3) If we don't remove the check, then we'd have to duplicate settle() code in _repay(). But if we change settle() in the future, we might overlook changing it here too.